tests/integration/test_sh_boot.py

```diff
@@ -345,3 +345,61 @@ def _check_pex_tools():
 
     # run the tools with seeded PEX_ROOT
     _check_pex_tools()
+
+
+@execution_mode
+def test_issue_2728_pex_python_and_pex_python_path(
+    tmpdir,  # type: Any
+    execution_mode_args,  # type: List[str]
+):
+    # type: (...) -> None
+    pex = os.path.realpath(os.path.join(str(tmpdir), "pex.sh"))
+
+    run_pex_command(args=["-o", pex] + execution_mode_args).assert_success()
+    pex_root = os.path.join(str(tmpdir), "pex_root")
+
+    def _check_pex_with_python():
+        for env in [dict(PEX_PYTHON="FIXME"), dict(PEX_PYTHON_PATH="FIXME")]:
+            output = subprocess.check_output(
+                args=[pex, "-c", "import sys; print(sys.version)"],
+                env=make_env(PEX_ROOT=pex_root, **env),
+                stderr=subprocess.STDOUT,
+            )
+            assert False, "FIXME"
```

Comment:
I am unclear on how to test these (and ran out of time today to investigate existing tests that use them): AIUI, the tests are run by tox with a specific interpreter available, and other may not be.

If there's a tip for how these should be structured, that'd be appreciated, otherwise I'll look at existing tests more, soon.